### DISCUSSION TITLE
3.0.0-alpha.0

### DISCUSSION THREAD

**Discussion by slorber** [facebook/docusaurus#9073] [Release feedback]
#### :boom: Breaking Change
* `create-docusaurus`, `docusaurus-tsconfig`
  * [#9050](https://github.com/facebook/docusaurus/pull/9050) feat: create official TypeScript base config @docusaurus/tsconfig ([@slorber](https://github.com/slorber))
* `create-docusaurus`, `docusaurus-mdx-loader`, `docusaurus-module-type-aliases`, `docusaurus-plugin-client-redirects`, `docusaurus-plugin-content-blog`, `docusaurus-plugin-content-docs`, `docusaurus-plugin-content-pages`, `docusaurus-plugin-debug`, `docusaurus-plugin-google-analytics`, `docusaurus-plugin-google-gtag`, `docusaurus-plugin-google-tag-manager`, `docusaurus-plugin-ideal-image`, `docusaurus-plugin-pwa`, `docusaurus-plugin-sitemap`, `docusaurus-preset-classic`, `docusaurus-theme-classic`, `docusaurus-theme-common`, `docusaurus-theme-live-codeblock`, `docusaurus-theme-mermaid`, `docusaurus-theme-search-algolia`, `docusaurus-types`, `docusaurus`
  * [#8961](https://github.com/facebook/docusaurus/pull/8961) feat: React 18 + automatic JSX runtime + build --dev ([@slorber](https://github.com/slorber))
* `create-docusaurus`
  * [#9026](https://github.com/facebook/docusaurus/pull/9026) chore: remove facebook template ([@slorber](https://github.com/slorber))
* `create-docusaurus`, `docusaurus-mdx-loader`, `docusaurus-migrate`, `docusaurus-plugin-content-blog`, `docusaurus-remark-plugin-npm2yarn`, `docusaurus-theme-classic`, `docusaurus-theme-common`, `docusaurus-theme-mermaid`, `docusaurus-types`, `docusaurus-utils-validation`, `docusaurus-utils`, `docusaurus`
  * [#8288](https://github.com/facebook/docusaurus/pull/8288) feat: upgrade to MDX v2 ([@slorber](https://github.com/slorber))
* `docusaurus-plugin-content-docs`, `docusaurus-theme-classic`, `docusaurus-theme-common`, `docusaurus`
  * [#7966](https://github.com/facebook/docusaurus/pull/7966) fix(plugin-docs,theme): refactor docs plugin routes and component tree ([@slorber](https://github.com/slorber))

<hr /><em>This discussion was created from the release <a href='https://github.com/facebook/docusaurus/releases/tag/v3.0.0-alpha.0'>3.0.0-alpha.0</a>.</em>

**Comment by lachieh** [comment]
Nice work, @slorber. I know this was a large amount of effort. 🎉

I notice there's mention of React 18's `lazy` and `Suspense` not working as expected. Is there anything you'd like feedback on or situations to look out for? We're currently making use of `react-loadable` in a couple of places and I'd like to swap that out so happy to experiment.

**Reply by slorber** [reply]
> Nice work, @slorber. I know this was a large amount of effort. 🎉

Thanks!

> I notice there's mention of React 18's lazy and Suspense not working as expected. Is there anything you'd like feedback on or situations to look out for? We're currently making use of react-loadable in a couple of places and I'd like to swap that out so happy to experiment.

It works on the client-side and SSR but it produces issues with hydration. It's not super clear what the behavior of Suspense should be for SSG: should we render it or not? If we render it how to ensure that the required lazy bundles are loaded as critical resources before we attempt to hydrate? With react-loadable it works because we collect those bundles to load before hydration, but we don't have such a solution for React.lazy

You can experiment if you want but I'm going to ask the React team anyway to have a correct answer on how to approach this problem.

**Reply by lachieh** [reply]
Thanks, that makes sense. I can see the divide. On one hand, it makes sense to put the loading work before the generative work so that everything loads on the client immediately, but on the other it makes the initial load on the client slightly larger.

I would be interested in following that conversation if there is a public place it is happening.

**Reply by slorber** [reply]
See also my comment here: https://github.com/facebook/docusaurus/issues/3841#issuecomment-1603022959

**Comment by Airkro** [comment]
Maybe `@docusaurus/react-loadable` needs to migrate to [@loadable/component](https://loadable-components.com/docs/loadable-vs-react-lazy/)?

**Reply by slorber** [reply]
We may do that indeed, or use another solution to drop react-loadable.

Why do you say "needs to migrate"? Why do you seem to consider it a mandatory part of v3.0?

See my comment here: https://github.com/facebook/docusaurus/issues/3841#issuecomment-1603022959

**Comment by nick-inkeep** [comment]
Congrats on the launch!

@slorber gave a shot at upgrading an existing V2 set.

Getting the following peer dependency warnings on docusaurus-maintained libraries:

warning "@docusaurus/preset-classic > @docusaurus/plugin-debug > react-json-view@1.21.3" has incorrect peer dependency "react@^17.0.0 || ^16.3.0 || ^15.5.4".
warning "@docusaurus/preset-classic > @docusaurus/plugin-debug > react-json-view@1.21.3" has incorrect peer dependency "react-dom@^17.0.0 || ^16.3.0 || ^15.5.4".
warning "@docusaurus/preset-classic > @docusaurus/plugin-debug > react-json-view > flux@4.0.4" has incorrect peer dependency "react@^15.0.2 || ^16.0.0 || ^17.0.0".
warning "@docusaurus/preset-classic > @docusaurus/theme-search-algolia > @docsearch/react > @algolia/autocomplete-preset-algolia@1.9.3" has unmet peer dependency "@algolia/client-search@>= 4.9.1 < 6".
warning "@docusaurus/preset-classic > @docusaurus/theme-search-algolia > @docsearch/react > @algolia/autocomplete-core > @algolia/autocomplete-plugin-algolia-insights@1.9.3" has unmet peer dependency "search-insights@>= 1 < 3".
warning "@docusaurus/preset-classic > @docusaurus/theme-search-algolia > @docsearch/react > @algolia/autocomplete-core > @algolia/autocomplete-shared@1.9.3" has unmet peer dependency "@algolia/client-search@>= 4.9.1 < 6".

Let me know how you'd like to triage.

**Reply by slorber** [reply]
The Docusaurus-related warnings have been fixed on https://github.com/facebook/docusaurus/pull/9116, the fix will be in alpha.1

I don't remember seeing the other warnings when initializing a new v3-alpha.0 site. Maybe it's a problem with your site dependencies and not Docusaurus? Let me know if you can't fix it

**Comment by kirtan403** [comment]
Very excited about it... Using docusaurus as my personal digital brain.. I see mdx v2 as one of the big features. Anything on the roadmap or anything I'm missing that is going on for v3? What's coming in v3? Any roadmap or anything?

**Reply by slorber** [reply]
v3 is mostly MDX 2 + React 18, among other smaller non-breaking change features

**Reply by bulleric** [reply]
is there already a migration guide from  2.4.x to 3.0.0?

**Reply by slorber** [reply]
No migration guide yet, but there are instructions in the PRs that have breaking changes (look at MDX 2 and React 18, those are the 2 major things that will be part of the migration guide)

- https://github.com/facebook/docusaurus/pull/8288
- https://github.com/facebook/docusaurus/pull/8961
